lib/src/client.dart

```diff
@@ -1157,4 +1157,77 @@ class TwitchClient {
     );
     return TwitchTokenRefresh.fromJson(data);
   }
+
+  /// Create a clip
+  ///
+  /// `broadcasterId` required
+  /// `hasDelay` required
+  ///
+  /// documentation : https://dev.twitch.tv/docs/api/reference/#create-clip
+  Future<TwitchCreateClipResponse> createClip({required String broadcasterId, required bool hasDelay}) async {
+    final data = await twitchHttpClient.postCall<Map<String, dynamic>>(
+      ['clips'],
+      <String, Object>{},
+      queryParameters: {'broadcaster_id': broadcasterId, 'has_delay': hasDelay},
+    );
+    return TwitchCreateClipResponse.fromJson(data);
+  }
+
+  /// Get a clip
+  ///
+  /// `broadcasterId` exclusive Required An ID that identifies the broadcaster
+  /// `gameId` exclusive Required An ID that identifies the game whose clips you want to get
+  /// `ids` exclusive Required An ID that identifies the clip to get
+  /// `startedAt` format RFC3339
+  /// `endedAt`   format RFC3339
+  /// `first` The maximum number of clips to return per page in the response.
+  /// `before` The cursor used to get the previous page of results
+  /// `after` The cursor used to get the next page of results
+  /// `isFeatured`
+  ///
+  /// documentation : https://dev.twitch.tv/docs/api/reference/#get-clips
+  Future<TwitchClipResponse> getClips({
+    String? broadcasterId,
+    String? gameId,
+    List<String>? ids,
+    String? startedAt,
+    String? endedAt,
+    int first = 20,
+    String? before,
+    String? after,
+    bool? isFeatured,
+  }) async {
+    assert(broadcasterId != null && gameId != null && ids != null, "One of exclusive parameter must have a value");
```

Comment:
Prefer this kind of check to ensure that only one parameter is specified and that the others are null:
```suggestion
    assert(
      switch ((broadcasterId, gameId, ids)) {
        (String(), null, null) => true,
        (null, String(), null) => true,
        (null, null, final ids?) when ids.isNotEmpty => true,
        _ => false,
      },
      'One of exclusive parameter must have a value',
    );
```

Reply:
Ok ok didn’t know we can do this pretty good syntax

Reply:
ok it didn't work as expected

Reply:
Yes we can, it's just that you need to bump the dependency constraints in `pubspec.yaml`:
```yaml
environment:
  sdk: ">=3.0.0 <4.0.0"
```